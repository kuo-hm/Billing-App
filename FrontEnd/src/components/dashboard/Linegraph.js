import { Box, Card, CardContent, CardHeader, Divider, useTheme } from "@mui/material";
import { useState } from "react";
import { Line } from "react-chartjs-2";

export const Linegraph = ({ chartData }) => {
  const theme = useTheme();
  const options = {
    responsive: true,
    interaction: {
      mode: "index",
      intersect: false,
    },

    stacked: false,
    plugins: {
      title: {
        display: false,
        text: "",
      },
    },
    scales: {
      y: {
        type: "linear",
        display: true,
        position: "left",
      },
    },
    animation: true,
    cornerRadius: 20,
    layout: { padding: 0 },
    legend: { display: false },
    maintainAspectRatio: false,
    responsive: true,
    xAxes: [
      {
        ticks: {
          fontColor: theme.palette.text.secondary,
        },
        gridLines: {
          display: false,
          drawBorder: false,
        },
      },
    ],
    yAxes: [
      {
        ticks: {
          fontColor: theme.palette.text.secondary,
          beginAtZero: true,
          min: 0,
        },
        gridLines: {
          borderDash: [2],
          borderDashOffset: [2],
          color: theme.palette.divider,
          drawBorder: false,
          zeroLineBorderDash: [2],
          zeroLineBorderDashOffset: [2],
          zeroLineColor: theme.palette.divider,
        },
      },
    ],
    tooltips: {
      backgroundColor: theme.palette.background.paper,
      bodyFontColor: theme.palette.text.secondary,
      borderColor: theme.palette.divider,
      borderWidth: 1,
      enabled: true,
      footerFontColor: theme.palette.text.secondary,
      intersect: false,
      mode: "index",
      titleFontColor: theme.palette.text.primary,
    },
  };

  const labels = [
    "January",
    "February",
    "March",
    "April",
    "May",
    "June",
    "July",
    "August",
    "September",
    "October",
    "November",
    "December",
  ];
  const data = {
    labels,
    datasets: [
      {
        label: "Devis With email sent",
        data: [
          chartData.devis_email_sent.january,
          chartData.devis_email_sent.february,
          chartData.devis_email_sent.march,
          chartData.devis_email_sent.april,
          chartData.devis_email_sent.may,
          chartData.devis_email_sent.june,
          chartData.devis_email_sent.july,
          chartData.devis_email_sent.august,
          chartData.devis_email_sent.september,
          chartData.devis_email_sent.october,
          chartData.devis_email_sent.november,
          chartData.devis_email_sent.december,
        ],
        borderColor: "rgb(255, 99, 132)",
        backgroundColor: "rgba(255, 99, 132, 0.5)",
        yAxisID: "y",
      },
      {
        label: "Devis validated",
        data: [
          chartData.devis_count_month.january,
          chartData.devis_count_month.february,
          chartData.devis_count_month.march,
          chartData.devis_count_month.april,
          chartData.devis_count_month.may,
          chartData.devis_count_month.june,
          chartData.devis_count_month.july,
          chartData.devis_count_month.august,
          chartData.devis_count_month.september,
          chartData.devis_count_month.october,
          chartData.devis_count_month.november,
          chartData.devis_count_month.december,
        ],
        borderColor: "rgb(53, 162, 235)",
        backgroundColor: "rgba(53, 162, 235, 0.5)",
        yAxisID: "y",
      },
      {
        label: "Factures validated",
        data: [
          chartData.factures_count_month.january,
          chartData.factures_count_month.february,
          chartData.factures_count_month.march,
          chartData.factures_count_month.april,
          chartData.factures_count_month.may,
          chartData.factures_count_month.june,
          chartData.factures_count_month.july,
          chartData.factures_count_month.august,
          chartData.factures_count_month.september,
          chartData.factures_count_month.october,
          chartData.factures_count_month.november,
          chartData.factures_count_month.december,
        ],
        borderColor: "rgb(53, 0, 235)",
        backgroundColor: "rgba(100, 162, 200, 0.5)",
        yAxisID: "y",
      },
    ],
  };
  const [selector, setSelector] = useState("");
  const handleChange = (event) => {
    setSelector(event.target.value);
  };
  return (
    <Card>
      <CardHeader
        // action={
        //   <FormControl variant="standard" sx={{ m: 1, minWidth: 120 }}>
        //     <InputLabel id="demo-simple-select-standard-label">Selector</InputLabel>
        //     <Select
        //       labelId="demo-simple-select-standard-label"
        //       id="demo-simple-select-standard"
        //       value={selector}
        //       onChange={handleChange}
        //       label="Age"
        //     >
        //       <MenuItem value={10}>By months</MenuItem>
        //       <MenuItem value={20}>By years</MenuItem>
        //     </Select>
        //   </FormControl>
        // }
        title="Chiffre D'affaire"
      />
      <Divider />
      <CardContent>
        <Box
          sx={{
            height: 500,
            position: "relative",
          }}
        >
          <Line data={data} options={options} />
        </Box>
      </CardContent>
      <Divider />
      {/* <Box
        sx={{
          display: 'flex',
          justifyContent: 'flex-end',
          p: 2
        }}
      >
        <Button
          color="primary"
          endIcon={<ArrowRightIcon fontSize="small" />}
          size="small"
        >
          Overview
        </Button>
      </Box> */}
    </Card>
  );
};
