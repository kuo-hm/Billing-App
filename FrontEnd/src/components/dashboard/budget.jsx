import MoneyIcon from "@mui/icons-material/Money";
import { Avatar, Card, CardContent, Grid, Typography } from "@mui/material";

export const Budget = ({ data }) => (
  <Card sx={{ height: "30vh" }}>
    <CardContent
      sx={{
        justifyContent: "space-between",
        display: "flex",
        flexDirection: "column",
        height: "30vh",
      }}
    >
      <Grid container spacing={3} sx={{ justifyContent: "space-between", flexDirection: "column" }}>
        <Grid item>
          <Typography color="textSecondary" gutterBottom variant="overline">
            Chiffre D affaire total
          </Typography>
          <Typography color="textPrimary" variant="h4">
            <>
            {data}DH</>
          </Typography>
        </Grid>
        <Grid item>
          <Avatar
            sx={{
              backgroundColor: "error.main",
              height: 56,
              width: 56,
            }}
          >
            <MoneyIcon />
          </Avatar>
        </Grid>
      </Grid>
    
    </CardContent>
  </Card>
);
