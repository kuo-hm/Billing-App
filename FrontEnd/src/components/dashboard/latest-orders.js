import {
  Box,
  Card,
  CardHeader,
  Checkbox,
  Table,
  TableBody,
  TableCell,
  TableHead,
  TablePagination,
  TableRow,
} from "@mui/material";
import { useState } from "react";
import PerfectScrollbar from "react-perfect-scrollbar";
import { SeverityPill } from "../severity-pill";

const LatestOrders = ({ text = "", data }) => {
  const [page, setPage] = useState(0);
  const [rowsPerPage, setRowsPerPage] = useState(5);
  const handleChangePage = (event, newPage) => {
    setPage(newPage);
  };

  const handleChangeRowsPerPage = (event) => {
    setRowsPerPage(parseInt(event.target.value, 10));
    setPage(0);
  };
  const emptyRows = rowsPerPage - Math.min(rowsPerPage, data.length - page * rowsPerPage);

  return (
    <Card>
      <CardHeader title={"Dernières " + text} />
      <PerfectScrollbar>
        <Box sx={{ minWidth: 800 }}>
          <Table>
            <TableHead>
              <TableRow>
                <TableCell>Order Ref</TableCell>
                <TableCell>Customer</TableCell>
                <TableCell>object</TableCell>

                <TableCell>Price</TableCell>
                <TableCell>Date</TableCell>
                <TableCell>Status</TableCell>
                <TableCell>email sent</TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {data.slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage).map((order) => (
                <TableRow hover key={order.id}>
                  <TableCell>{order.id}</TableCell>
                  <TableCell>{order.customer}</TableCell>
                  <TableCell>{order.object}</TableCell>
                  <TableCell>{order.prix}</TableCell>
                  <TableCell>{order.creation_date}</TableCell>
                  <TableCell>
                    <SeverityPill
                      color={
                        (order.status === "1" && "success") ||
                        (order.status === "2" && "warning") ||
                        "warning"
                      }
                    >
                      {order.status === "1"
                        ? "Payé"
                        : order.status === "2"
                        ? "Non payé"
                        : "non validé"}
                    </SeverityPill>
                  </TableCell>
                  <TableCell>
                    <Checkbox defaultChecked checked={order.email_sent} />
                  </TableCell>
                </TableRow>
              ))}
            </TableBody>
          </Table>
          <TablePagination
            rowsPerPageOptions={[5, 10, 25]}
            component="div"
            count={data.length}
            rowsPerPage={rowsPerPage}
            page={page}
            onPageChange={handleChangePage}
            onRowsPerPageChange={handleChangeRowsPerPage}
          />
        </Box>
      </PerfectScrollbar>
    </Card>
  );
};

export default LatestOrders;
