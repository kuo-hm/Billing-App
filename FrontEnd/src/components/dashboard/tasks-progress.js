import InsertChartIcon from "@mui/icons-material/InsertChartOutlined";
import { Avatar, Card, CardContent, Grid, Typography } from "@mui/material";

export const TasksProgress = ({ data }) => (
  <Card sx={{ height: "30vh" }}>
    <CardContent
      sx={{
        justifyContent: "space-between",
        display: "flex",
        flexDirection: "column",
        justifyContent: "space-between",
        height: "30vh",
      }}
    >
      <Grid container spacing={3} sx={{ justifyContent: "space-between", flexDirection: "column" }}>
        <Grid item>
          <Typography color="textSecondary" gutterBottom variant="overline">
            DEVIS VALIDÃ‰S
          </Typography>
          <Typography color="textPrimary" variant="h4">
            {data}
          </Typography>
        </Grid>
        <Grid item>
          <Avatar
            sx={{
              backgroundColor: "warning.main",
              height: 56,
              width: 56,
            }}
          >
            <InsertChartIcon />
          </Avatar>
        </Grid>
      </Grid>

    </CardContent>
  </Card>
);
