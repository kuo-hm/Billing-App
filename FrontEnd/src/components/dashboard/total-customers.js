import PeopleIcon from "@mui/icons-material/PeopleOutlined";
import { Avatar, Card, CardContent, Grid, Typography } from "@mui/material";

export const TotalCustomers = ({ data }) => (
  <Card>
    <CardContent
      sx={{
        justifyContent: "space-between",
        display: "flex",
        flexDirection: "column",
        justifyContent: "space-between",
        height: "30vh",
      }}
    >
      <Grid container spacing={3} sx={{ justifyContent: "space-between", flexDirection: "column" }}>
        <Grid item>
          <Typography color="textSecondary" gutterBottom variant="overline">
            CUSTOMERS
          </Typography>
          <Typography color="textPrimary" variant="h4">
            {data}
          </Typography>
        </Grid>
        <Grid item>
          <Avatar
            sx={{
              backgroundColor: "success.main",
              height: 56,
              width: 56,
            }}
          >
            <PeopleIcon />
          </Avatar>
        </Grid>
      </Grid>
    </CardContent>
  </Card>
);
