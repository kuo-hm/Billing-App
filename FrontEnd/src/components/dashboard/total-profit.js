import AttachMoneyIcon from "@mui/icons-material/AttachMoney";
import { Avatar, Card, CardContent, Grid, Typography } from "@mui/material";

export const TotalProfit = ({ data }) => (
  <Card>
    <CardContent
      sx={{
        justifyContent: "space-between",
        display: "flex",
        flexDirection: "column",
        justifyContent: "space-between",
        height: "30vh",
      }}
    >
      <Grid container spacing={3} sx={{ justifyContent: "space-between", flexDirection: "column" }}>
        <Grid item>
          <Typography color="textSecondary" gutterBottom variant="overline">
            Factures pay√©es ce mois-ci
          </Typography>
          <Typography color="textPrimary" variant="h4">
            {data} DH
          </Typography>
        </Grid>
        <Grid item>
          <Avatar
            sx={{
              backgroundColor: "primary.main",
              height: 56,
              width: 56,
            }}
          >
            <AttachMoneyIcon />
          </Avatar>
        </Grid>
      </Grid>
    </CardContent>
  </Card>
);
