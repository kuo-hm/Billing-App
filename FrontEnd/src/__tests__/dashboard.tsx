import { render } from '@testing-library/react'
import { Budget } from '../components/dashboard/budget'

it('renders homepage unchanged', () => {
    const { container } = render(<Budget data={55} />)
  expect(container).toMatchSnapshot()
})