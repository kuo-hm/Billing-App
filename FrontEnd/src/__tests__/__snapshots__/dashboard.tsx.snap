// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-iohzb7-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-1doetlm-MuiCardContent-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-rmhki7-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <span
            class="MuiTypography-root MuiTypography-overline MuiTypography-gutterBottom css-chjbjb-MuiTypography-root"
          >
            Chiffre D affaire total
          </span>
          <h4
            class="MuiTypography-root MuiTypography-h4 css-wz9of0-MuiTypography-root"
          >
            55
            DH
          </h4>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-101pa6z-MuiAvatar-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="MoneyIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M5 8h2v8H5zm7 0H9c-.55 0-1 .45-1 1v6c0 .55.45 1 1 1h3c.55 0 1-.45 1-1V9c0-.55-.45-1-1-1zm-1 6h-1v-4h1v4zm7-6h-3c-.55 0-1 .45-1 1v6c0 .55.45 1 1 1h3c.55 0 1-.45 1-1V9c0-.55-.45-1-1-1zm-1 6h-1v-4h1v4z"
              />
              <path
                d="M2 4v16h20V4H2zm2 14V6h16v12H4z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
